File: 1441_build_an_array_with_stack_operations/Solution.java
// LeetCode 1441 - Build An Array With Stack Operations
// https://leetcode.com/problems/build-an-array-with-stack-operations/

import java.util.*;

class Solution {
    public List<String> buildArray(int[] target, int n) {
        var answer = new ArrayList<>(); int current = 1;
        for (int value : target) {
            while (current < value) { answer.add("Push"); answer.add("Pop"); current++; }
            answer.add("Push"); current++;
        }
        return answer;
    }
}

File: 1441_build_an_array_with_stack_operations/Solution.swift
// LeetCode 1441 - Build an Array With Stack Operations
// https://leetcode.com/problems/build-an-array-with-stack-operations/

class Solution {
    func buildArray(_ target: [Int], _ n: Int) -> [String] {
        var answer = [String](), current = 1
        for value in target {
            while current < value {
                answer.append("Push"); answer.append("Pop"); current += 1
            }
            answer.append("Push"); current += 1
        }
        return answer
    }
}
